### DISCUSSION TITLE
How to disable WSL on windows-latest/windows-2019 for Github Action

### DISCUSSION THREAD

**Discussion by zdtsw** [actions/virtual-environments#6049] [General questions]
Background:
we used to use windows-2016 in our github action workflows, till recently realized it is not supported any more. thus, upgraded it to windows-2019/windows-latest.

what the action job is configed to run: 
- on windows install "cygwin" and a bunch of tools (e.g make, git, antcontrib, etc) 
- run "make" which in turn to call  "ant" build in order to run java tests. 
-  one of the target in the "ant" is to do "git clone" some extra repo.

I believe it was working on windows-2016 till now on windows-2019/latest.
the error I got when the line of "git clone" hit is:
```
     [exec] Windows Subsystem for Linux has no installed distributions.
     [exec] Distributions can be installed by visiting the Microsoft Store:
     [exec] https://aka.ms/wslstore
```

I am only guessing, this is because wsl is messed up with cygwin, so if i am able to disable wsl by pwsh as the first step, then cygwin and so on should still work.
(or maybe I am totally wrong.....)
just wanna see if anyone has idea how to fix this.

P.S: I've tried to remove the customized logic (to install cygwin, git, make, ant etc) just let windows-2019 wls handle everything as [stated](https://github.com/actions/runner-images/blob/main/images/win/Windows2019-Readme.md?rgh-link-date=2022-03-23T16%3A49%3A14Z)
but then the path seems also mixed up with both "\" and "/"

P.S2: I've also tried with 
```
- name: Disable WSLv1
      run: |
        Disable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux
      shell: pwsh
```
but does not work as well....

**Comment by zdtsw** [comment]
I have also tried to use
```
    - uses: Vampire/setup-wsl@v1
      with:
        distribution: Ubuntu-20.04
        additional-packages:
          ant
          make
```
seems the error of 
```      
     [exec] Windows Subsystem for Linux has no installed distributions.
     [exec] Distributions can be installed by visiting the Microsoft Store:
     [exec] https://aka.ms/wslstore
```   i
s gone, but still might be good to know what is the best way to handle these windows instances since 2019+ became the only available ones

**Comment by sogaiu** [comment]
Rather late to the party, but I encountered a similar issue where it appeared that `bash` would cause `c:\windows\system32\bash.exe` to execute [1], leading to what looked like the "wsl triggering" described above.

At least, that's what [this answer](https://superuser.com/a/1841536) to the question "Why executing 'bash' in the terminal opens wsl on Windows 10?" seemed to suggest [2].

I didn't find a good way to disable WSL (nor convince the runner via setting `PATH` to stay away from `c:\windows\system32\bash.exe`), but thanks to [this idea](https://stackoverflow.com/a/69251820) by @Mizux, I was "inspired" to remove `c:\windows\system32\bash.exe`.

That was enough to get my job to execute appropriately [3].  Perhaps this doesn't count as disabling WSL, but may be it will help someone else in a similar situation.

---

[1] Within the context of some other code that was "shelling out" to `bash`.

[2] Thanks to [Andrii Hnatushchenko](https://superuser.com/users/1908990/andrii-hnatushchenko) for asking and [Tonny](https://superuser.com/users/83250/tonny) for that helpful answer.

[3] Additional thanks to [iirekm](https://stackoverflow.com/users/469153/iirekm) for articulating [this](https://stackoverflow.com/a/59988803).  It nicely crystalized what I seemed to already be trying to do (^^;

**Reply by felipec** [reply]
But `C:/Windows/System32/bash.exe` is a system file, you can't remove it.

**Reply by sogaiu** [reply]
Hmm, I think it worked for me.

May be it doesn't work any more?

**Comment by felipec** [comment]
I think the answer is that it cannot be disabled at run time. At least according to: [this comment](https://github.com/actions/runner-images/issues/7253#issuecomment-1500121978).
